ovc/client.go

```diff
@@ -230,47 +285,90 @@ func (c *Client) Do(req *http.Request) ([]byte, error) {
 			defer taskResp.Body.Close()
 		}
 		if err != nil {
+			c.logger.Errorf("Error getting task result: %s", err)
+			if requestErrorCount < 20 {
+				time.Sleep(2 * time.Second)
+				requestErrorCount++
+				continue
+			} else {
+				c.logger.Errorf("Could not get task result: %s", err)
+				return nil, err
+			}
+		}
+
+		c.logger.Debugf("OVC call: %s", req.URL.Path)
+		c.logger.Debugf("OVC task call: %s", taskID)
+		c.logger.Debugf("OVC response status code: %s", taskResp.StatusCode)
+		c.logger.Debugf("OVC response status: %s", taskResp.Status)
+		resultBody, err := ioutil.ReadAll(taskResp.Body)
+		if err != nil {
+			c.logger.Errorf("Could not read response body: %s", err)
 			return nil, err
 		}
+		c.logger.Debugf("OVC response: %s", string(resultBody))
+
 		switch {
 		case taskResp.StatusCode == 401:
+			c.logger.Errorf("Unauthorized: %s", ErrAuthentication)
 			return nil, ErrAuthentication
 		case taskResp.StatusCode == 404:
-			// task may have not been registered yet
+			if fourOfourCount == 0 {
+				fourOfourCount++
+				c.logger.Error("Oops we hit a race condition bug in the API server prior 2.5.6")
+				time.Sleep(2 * time.Second)
+				continue
+			} else {
+				c.logger.Errorf("Task not found: %s", ErrNotFound)
+				return nil, ErrNotFound
+			}
+		case taskResp.StatusCode == 400:
+			// Sometimes nginx returns 400 for no reason
+			c.logger.Error("Received 400, probably nginx issue.")
+			time.Sleep(2 * time.Second)
+			continue
```

Comment:
If for whatever reason the server side is misbehaving and consistently returning a `400` we never get out of the loop

Reply:
i solved this by moving the timeout to the beginning of the loop